lib/sensors/accelerometer.cpp

```diff
@@ -85,37 +85,37 @@ std::optional<std::int16_t> Accelerometer::getRawAccelerationZ()
   return raw_acceleration_Z;
 }
 
-// TODOLater: !!!!! this code couldn't be tested as the sensor wasn't working !!!!!!!!
+// todolater: check whether the range of values the accelerometer is correct, cause now the noise is big
 std::optional<core::RawAccelerationData> Accelerometer::read()
 {
   /* check to see if the values are ready to be read */
-  auto dataReady = i2c_.readByte(mux_address_, kDataReady);
+  auto dataReady = i2c_.readByte(kDeviceAddr, kDataReady);
   if (!dataReady) {
     log_.log(core::LogLevel::kFatal, "acceleration data could not be read");
     return std::nullopt;
   }
   if (dataReady.value() % 2 == 0) {
-    log_.log(core::LogLevel::kFatal, "acceleration data not ready yet to be read");
+    log_.log(core::LogLevel::kInfo, "acceleration data not ready yet to be read");
     return std::nullopt;
   }
 
   // x axis
   auto resultX = getRawAccelerationX();
   if (!resultX) return std::nullopt;
   const std::int16_t XRawAcc = resultX.value(); 
-  core::Float XAcceleration = (static_cast<core::Float>(XRawAcc)) * 0.488f;
+  int16_t XAcceleration = (int16_t) ((((int32_t) XRawAcc) * 488) / 1000);
```

Comment:
`static_cast` was good

Reply:
also magic numbers

Reply:
turns out the static cast to a float was the whole reason why the accelerometer didn't work. Hence, I am sticking to int here.

Reply:
i don't care what you static cast to, but you should static cast!

`static_cast<std::int16_t>(...)`